app/src/Containers/EventsContainer.tsx

```diff
@@ -23,29 +31,109 @@ const HorizontalLine = styled.View`
   margin: 0px 75px 10px 75px;
 `
 
-const EventList: FC<EventProps> = ({ data }) => {
-  return (
-    <SafeArea>
-      <TopOfApp />
-      <EventSearch />
-      <EventOptions />
-      <HorizontalLine />
-      <EventReturnedList data={data} />
-    </SafeArea>
-  )
-}
+const SearchResultsContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  margin-bottom: 20px;
+  width: 100%;
+`
 
-const EventsContainer = () => {
-  const [fetchAllEvents, { data: events }] = useLazyFetchAllEventsQuery()
+const SearchResultsLabel = styled.Text`
+  color: ${props => props.theme.colors.staBlack};
+  text-align: center;
+  width: 100%;
+`
 
+const EventsContainer = (props: {
+  route: {
+    params: {
+      selectedOption?: EventsRange | 'myEvents'
+      search: EventSearchInterface
+    }
+  }
+}) => {
+  const [fetchAllUpcomingEvents, { data: allUpcomingEvents }] =
+    useLazyFetchAllUpcomingEventsQuery()
+  const dispatch = useDispatch()
+  const [eventsSearch, setEventsSearch] = useState<
+    EventSearchInterface | undefined
+  >()
+  const [selectedOption, setSelectedOption] = useState<
+    EventsRange | 'myEvents'
+  >(EventsRange.Upcoming)
+
+  // When the component is first created...
   useEffect(() => {
-    fetchAllEvents('')
-  }, [fetchAllEvents])
+    // Get all upcoming events from the API
+    fetchAllUpcomingEvents('')
+  }, [])
+
+  // When allUpcomingEvents is set...
+  useEffect(() => {
+    // Store all upcoming events in the Redux store so they can be used by other components too e.g. EventSearchContainer
+    if (allUpcomingEvents) {
+      dispatch(setEvents({ upcoming: allUpcomingEvents }))
+    }
+  }, [allUpcomingEvents])
+
+  // When the user changes search options or they tap Past/Upcoming/My events navigation occurs,
+  // this changes the route parameters - we use this to update EventOptions and to work out
+  // whether to show events search results or all events in the list
+  useEffect(() => {
+    setSelectedOption(
+      props.route.params?.selectedOption ?? EventsRange.Upcoming,
+    )
+    setEventsSearch(props.route.params?.search)
+  }, [props.route.params])
+
+  const EventList: FC<EventProps> = ({ data }) => {
+    return (
+      <SafeArea>
+        <TopOfApp />
+        <EventSearch />
+        <EventOptions selected={EventsRange.Upcoming} />
+
+        {/* If the user has done a quick search for upcoming events, show those
+            quick search buttons so they can amend their quick search if they want,
+            without having to go back to the search screen */}
```

Comment:
From the current Figma designs, it looks like these quick search buttons should also appear on the results screen when they've been used for a search